pkg/piecestore/rpc/server/retrieve.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package server
+
+import (
+	"errors"
+	"io"
+	"log"
+	"os"
+
+	"storj.io/storj/pkg/piecestore"
+	"storj.io/storj/pkg/utils"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+// Retrieve -- Retrieve data from piecestore and send to client
+func (s *Server) Retrieve(stream pb.PieceStoreRoutes_RetrieveServer) error {
+	log.Println("Retrieving data...")
+
+	// Receive Signature
+	recv, err := stream.Recv()
+	if err != nil || recv == nil {
+		log.Println(err)
+		return errors.New("Error receiving Piece data")
+	}
+
+	pd := recv.GetPieceData()
+	log.Printf("ID: %s, Size: %v, Offset: %v\n", pd.GetId(), pd.GetSize(), pd.GetOffset())
+
+	// Get path to data being retrieved
+	path, err := pstore.PathByID(pd.GetId(), s.DataDir)
+	if err != nil {
+		return err
+	}
+
+	// Verify that the path exists
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return err
```

Comment:
do you want to put logging statements if the path doesn't exist?

Reply:
not if you're returning an error, but this error should probably be wrapped. the stdlib won't attach a stack trace, so wrapping will